cuda_core/cuda/core/experimental/_memory/_buffer.pyx

```diff
@@ -419,6 +382,36 @@ cdef inline void Buffer_close(Buffer self, stream):
         self._alloc_stream = None
 
 
+cdef inline void Buffer_fill_uint8(Buffer self, uint8_t value, cydriver.CUstream s):
+    with nogil:
+        HANDLE_RETURN(cydriver.cuMemsetD8Async(<cydriver.CUdeviceptr>self._ptr, value, self._size, s))
```

Comment:
nit: we want the exception clause `except*` since `HANDLE_RETURN` can raise, but we don't want Cython to warn, so with Cython 3 we general want to avoid using `void` as the return type and use this instead:
```suggestion
cdef inline int Buffer_fill_uint8(Buffer self, uint8_t value, cydriver.CUstream s) except?-1:
    with nogil:
        HANDLE_RETURN(cydriver.cuMemsetD8Async(<cydriver.CUdeviceptr>self._ptr, value, self._size, s))
    return 0
```